server/src/main/java/org/eclipse/openvsx/search/SearchService.java

```diff
@@ -188,10 +188,12 @@ else if (value > 1.0)
             queryBuilder.withFilter(QueryBuilders.matchPhraseQuery("categories", category));
         }
 
-        // Configure default sorting of results
-        queryBuilder.withSort(SortBuilders.scoreSort());
-        queryBuilder.withSort(SortBuilders.fieldSort("relevance").unmappedType("float").order(SortOrder.DESC));
-
+        if ("relevance".equals(sortBy)) {
+            queryBuilder.withSort(SortBuilders.scoreSort());
+        }
+        queryBuilder
+                .withSort(SortBuilders.fieldSort(sortBy).unmappedType("float").order(SortOrder.fromString(sortOrder)));
```

Comment:
Please check the `sortBy` value, e.g. with a `switch` statement, and give a proper error message if it doesn't match any predefined value.

`unmappedType("float")` is not correct for all fields. See `ExtensionSearch` to see the field definitions.